lib/trivial.nix

```diff
@@ -467,6 +471,41 @@ rec {
     then v
     else k: v;
 
+  /*
+    Make a function lazy in its arguments.
+
+    Normally when you call a function like `{ a, b }: { r = foo a b; }`,
+    the attribute names of the argument must be evaluated before the evaluator
+    can return a the (partially evaluated) function body (`{ r = ...; }`).
+
```

Comment:
```suggestion
    can return a the (partially evaluated) function body (`{ r = ...; }`).
    `lazyFunction` can change such a function to not evaluate the attribute
    names of the argument before the body is returned.

```

Reply:
I've rewritten a lot to make better use of the example instead.